### PR TITLE
docs: document proxy protocol using sequence diagram

### PR DESCRIPTION
**Pull request by jcvenegas**
This PR adds a new sequence diagram that
shows the message flows for `create` and `start` commands between the following:
- proxy
- shim
- runtime
- agent

Fixes: #104

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: docs/how-to-generate-diagrams.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#How to generate diagrams
+
+Proxy diagrams should be generated by preference with Plantuml.
```
Comment (jodh-intel) [inline docs/how-to-generate-diagrams.md]: According to their website, they seem to prefer the name to be capitalised as "plantUML".

File: docs/how-to-generate-diagrams.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+Plantuml is a tool that allows to generate diagrams in different image formats
```
Comment (jodh-intel) [inline docs/how-to-generate-diagrams.md]: I'd consider shortening this to, "PlantUML is a tool that generates UML in different image formats from a textual description".

File: docs/how-to-generate-diagrams.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+using a text description.
+
+In you can generate diagrams by its web server: http://www.plantuml.com/plantuml/uml
```
Comment (jodh-intel) [inline docs/how-to-generate-diagrams.md]: I'd make this fully impersonal. Something like, "Diagrams can be generated directly in a browser at http://www.plantuml.com/plantuml/uml, or locally at http://plantuml.com/download."

File: docs/proxy-protocol-sequence-diagram.txt [hunk 4]
```diff
@@ -0,0 +1,66 @@
+@startuml
+'Clear Containers Proxy Protocol sequence digram
```
Comment (jodh-intel) [inline docs/proxy-protocol-sequence-diagram.txt]: - "Clear Container" => "Intel® Clear Containers".
- Typo and missing period: "digram" => "diagram."

File: docs/proxy-protocol-sequence-diagram.txt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+'This is the textual representation of a sequence diagram
```
Comment (jodh-intel) [inline docs/proxy-protocol-sequence-diagram.txt]: Missing period?

**Comment by jodh-intel** [comment]
This looks great! Just a few minor nits on first pass.

**Comment by jcvenegas** [comment]
@jodh-intel  thanks for the comments, suggested changes applied.

**Comment by jcvenegas** [comment]
hi @iphutch, could you please take a look to this PR?

File: docs/how-to-generate-diagrams.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#How to generate diagrams
```
Comment (iphutch) [inline docs/how-to-generate-diagrams.md]: Missing a space after the #

File: docs/how-to-generate-diagrams.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+
+Proxy diagrams should be generated by preference with PlantUML.
```
Comment (iphutch) [inline docs/how-to-generate-diagrams.md]: Let's provide an embedded link for the first instance of PlantUML:
[PlantUML](http://plantuml.com)

**Comment by jodh-intel** [comment]
Once @iphutch's comments are addressed,

lgtm

**Comment by jcvenegas** [comment]
@iphutch thanks, changes requested were applied.

**Comment by sameo** [comment]
LGTM

**Pull request merged**